system-tests/systemTestUtils.js

```diff
@@ -1,47 +1,73 @@
 'use strict';
 
+const _ = require('lodash');
+const fs = require('fs');
+const os = require('os');
 const path = require('path');
-const stripAnsi = require('strip-ansi');
+const { promisify } = require('util');
 
-function caseFilePath(caseNumber, fileName) {
+const replaceBackslashes = require('../lib/testUtils/replaceBackslashes');
+
+const copyFile = promisify(fs.copyFile);
+const readFile = promisify(fs.readFile);
+
+function caseFilePath(caseNumber, fileName = 'stylesheet') {
 	return replaceBackslashes(path.join(__dirname, caseNumber, fileName));
 }
 
-function caseStylesheetGlob(caseNumber) {
+function caseFiles(caseNumber) {
 	return caseFilePath(caseNumber, 'stylesheet.*');
 }
 
-function caseConfig(caseNumber, ext = 'json') {
+function caseConfigFile(caseNumber, ext = 'json') {
 	return caseFilePath(caseNumber, `config.${ext}`);
 }
 
-function prepResults(results) {
-	return results.map((result) => {
-		// The _postcssResult object is not part of our API and is huge
-		const { _postcssResult, source, ...preppedResult } = result;
-
-		// The `source` of each file will not be the same on different machines or platforms
-		preppedResult.source = path
-			.relative(__dirname, source)
-			.replace(new RegExp(`\\${path.sep}`, 'g'), '/');
+async function caseConfig(caseNumber) {
+	return JSON.parse(await readFile(caseConfigFile(caseNumber, 'json'), 'utf8'));
+}
 
-		return preppedResult;
-	});
+async function caseCode(caseNumber, ext = 'css') {
+	return await readFile(`${caseFilePath(caseNumber)}.${ext}`, 'utf8');
 }
 
-function replaceBackslashes(str) {
-	return str.replace(/\\/g, '/');
+async function caseFilesForFix(caseNumber, ext = 'css') {
+	const tempPath = replaceBackslashes(path.join(os.tmpdir(), `stylesheet-${_.uniqueId()}.${ext}`));
+
+	await copyFile(path.join(__dirname, caseNumber, `stylesheet.${ext}`), tempPath);
+
+	return tempPath;
 }
 
-function stripColors(input) {
-	return stripAnsi(input);
+function prepForSnapshot({ results, output, ...rest }) {
+	// If output isn't fixed code
+	if (output.startsWith('[')) {
+		// The `source` of each file varies between platforms or if a tmp file is used
+		output = JSON.parse(output).map((warning) => {
+			delete warning.source;
+
+			return warning;
+		});
+	}
+
+	return {
+		// The _postcssResult object is not part of our API and is huge
+		results: results.map(({ _postcssResult, ...rest }) => {
+			delete rest.source;
+
+			return rest;
+		}),
+		output,
+		...rest,
```

Comment:
For safety let's use different names for `rest`, so we're not confused in `results` scope what `rest` we're dealing with.

Also we can use single way of removing unneeded property:

```js
		results: results.map(({ _postcssResult, source, ...rest }) => {
			return rest;
		}),
```

Or:

```js
		results: results.map((result) => {
			delete result.source;
			delete result._postcssResult;

			return result;
		}),
```